### PR TITLE
HashCalc: Perf: use refs for hashes

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
Avoid copying hash values. Store refs instead. Perf is > 10%.
#### Summary of Changes

Fixes #

### CODE REVIEW

File: runtime/src/accounts_hash.rs [hunk 1]
```diff
@@ -676,7 +677,7 @@ impl AccountsHash {
                 overall_sum = Self::checked_cast_for_capitalization(
                     item.lamports as u128 + overall_sum as u128,
                 );
-                hashes.push(item.hash);
+                hashes.push(&item.hash);
```
Comment (jeffwashington) [inline runtime/src/accounts_hash.rs:680]: this is the root change

File: runtime/src/accounts_hash.rs [hunk 2]
```diff
@@ -1449,34 +1459,34 @@ pub mod tests {
...
     }
+    /*
+        fn test_de_dup_accounts_in_parallel(
+            account_maps: &[CalculateHashIntermediate],
+        ) -> (Vec<&Hash>, u64, usize) {
+            AccountsHash::default()
+                .de_dup_accounts_in_parallel(&vec![vec![account_maps.to_vec()]][..], 0)
+        }
+        #[test]
+        fn test_accountsdb_remove_zero_balance_accounts() {
```
Comment (brooksprumo) [inline runtime/src/accounts_hash.rs:1470]: Should these tests be commented out? Are they broken?

**Review by brooksprumo — COMMENTED** [review]
Love it. I was always curious why `Hash` was marked with `Copy`...

File: runtime/src/accounts_hash.rs (hunk 2 above)
Comment (jeffwashington) [reply runtime/src/accounts_hash.rs:1470]: Doh. nope. changed signatures and was trying to see how fast it would run. I'll fix that.

Comment (jeffwashington) [reply runtime/src/accounts_hash.rs]: removed the /*... and fixed the test and function.

**Review by brooksprumo — APPROVED** [review]

**Pull request merged**